R/type-vctr.R

```diff
@@ -475,17 +475,27 @@ quantile.vctrs_vctr <- function(x, ..., type = 1, na.rm = FALSE) {
   # nocov end
 }
 
+vec_cast_or_na <- function(x, to, ...) {
+  tryCatch(
+    vctrs_error_incompatible_type = function(...) vec_cast(rep(NA, length(x)), to),
+    vec_cast(x, to)
+  )
+}
+
 #' @export
 min.vctrs_vctr <- function(x, ..., na.rm = FALSE) {
   if (vec_is_empty(x)) {
-    return(vec_cast(Inf, x))
+    return(vec_cast_or_na(Inf, x))
   }
 
   # TODO: implement to do vec_arg_min()
   rank <- xtfrm(x)
 
   if (isTRUE(na.rm)) {
     idx <- which.min(rank)
+    if (vec_is_empty(idx)) {
```

Comment:
Are the empty idx cases covered by unit tests?

Reply:
I don't think so, I'll add in some extra tests. Thanks!